pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -178,21 +181,53 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 	return utilerrors.NewAggregate(errs)
 }
 
+// validateCertificateSAN makes sure that a given certificate is valid
+// at least for the SANs defined in the configuration.
+func (c *EtcdCertSignerController) validateCertificateSAN(certName string, ipAddresses []string) error {
+	secret, err := c.secretLister.Secrets(operatorclient.TargetNamespace).Get(certName)
+	if err != nil {
+		return fmt.Errorf("could not get secrets from %s: %w", certName, err)
+	}
+	if secret.Data == nil {
+		return fmt.Errorf("secret data not found in %s", certName)
+	}
+	certificates, err := cert.ParseCertsPEM(secret.Data["tls.crt"])
+	if err != nil {
+		return fmt.Errorf("could not parse TLS cert for %s: %w", certName, err)
+	}
+	// Using the first certificate to verify the IP addresses
+	certificate := certificates[0]
+	for _, ipAddress := range ipAddresses {
+		if err := certificate.VerifyHostname(ipAddress); err != nil {
+			return fmt.Errorf("SAN for the certificate %s does not include %s: %w", certName, ipAddress, err)
+		}
+	}
+	return nil
+}
+
 func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node, recorder events.Recorder) error {
 	etcdPeerClientCertName := tlshelpers.GetPeerClientSecretNameForNode(node.Name)
 	etcdServingCertName := tlshelpers.GetServingSecretNameForNode(node.Name)
 	metricsServingCertName := tlshelpers.GetServingMetricsSecretNameForNode(node.Name)
 
 	var err error
```

Comment:
is this useful? why not be explicit about `err` scope. seems like some stuff can get cleaned up below